hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/store/BaseHeapNearCacheRecordStore.java

```diff
@@ -163,4 +163,26 @@ public R apply(K key, R reservedRecord) {
         Object cachedValue = existingRecord.getValue();
         return cachedValue instanceof Data ? toValue(cachedValue) : (V) cachedValue;
     }
+
+    @Override
+    public void invalidate(K key) {
+        records.applyIfPresent(key, invalidatorFunction);
```

Comment:
Minor: I know the OS side doesn't do anything but can we call the `onRemove` and `onRemoveError` here for consistency with EE?